### PR TITLE
Refactor CMakeLists to support Yocto build

### PR DESCRIPTION
**Pull request by cajun-rat**
This includes several changes to support building in Yocto
 - Both RVI and SWM support are now behind a single BUILD_GENIVI flag
 - All third_party dependencies are build with cmake
 - A new flag disables building tests, to avoid a dependency on gtest/gmock

### CODE REVIEW

**Comment by cajun-rat** [comment]
@taheris Can you re-check this please?

**Comment by patriotyk** [comment]
Looks good to me, but do we really need session.conf file?

**Comment by cajun-rat** [comment]
@patriotyk On my Ubuntu desktop, I had trouble with dbus permission errors without it.  I don't really know enough dbus, but I _think_ that dbus-run-session uses the config file that happens to be installed in the system by default.  Isolating the tests from that seemed to be a good idea, but I'm no expert.

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -53,36 +34,33 @@ if(BUILD_WITH_CODE_COVERAGE)
...
 
 if(CLANG_FORMAT)
-add_custom_target(format-tools
-    COMMAND ${CLANG_FORMAT} -i -style file ${CMAKE_CURRENT_SOURCE_DIR}/src/*.cc ${CMAKE_CURRENT_SOURCE_DIR}/src/*.h ${CMAKE_CURRENT_SOURCE_DIR}/tests/*.cc
-    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-    COMMENT "Running clang-format"
-    VERBATIM)
-add_dependencies(qa format-tools)
+    add_custom_target(format-tests
+        COMMAND ${CLANG_FORMAT} -i -style=file ${ALL_TEST_SRCS}
```
Comment (patriotyk) [inline tests/CMakeLists.txt]: With this change, You have not provided target for formatting sources, only tests.

**Comment by taheris** [comment]
lgtm

File: tests/CMakeLists.txt (hunk 1 above)
Comment (cajun-rat) [inline tests/CMakeLists.txt]: I've put a separate clang-format invocation in src/CMakeLists.txt, too.  This means we call it twice, but it doesn't have to reference files between the two directories.  Does that make sense?

**Comment by patriotyk** [comment]
Found one bug. all executables placed in root folder instead target

File: tests/CMakeLists.txt (hunk 1 above)
Comment (patriotyk) [inline tests/CMakeLists.txt]: I think it is ok. But why did You split it?

**Comment by cajun-rat** [comment]
@patriotyk What is the advantage of putting them in the 'target' subfolder?

File: tests/CMakeLists.txt (hunk 1 above)
Comment (cajun-rat) [inline tests/CMakeLists.txt]: The problem is that clang-format needs a list of files to operate on. Globbing for these seems a little ugly, and so the two options seem to be:
1) Pass the list of files between the src/ and test/ CMakeLists.txt 
2) Run clang-format twice

My thought is that keeping the CMakeLists.txt self-contained such that they only operate on files in their folders is a bit nicer?

**Comment by cajun-rat** [comment]
Oh, do you normally make 'in-tree' builds?  I normally create a separate build output directory (like 'build') and run "cd build; cmake .."

**Comment by patriotyk** [comment]
Yes I do.

**Comment by cajun-rat** [comment]
Ah OK.  Would switching to out-of-tree builds (or having the build output in the project root directory) be a problem?

**Comment by patriotyk** [comment]
No, all tests pass, I just thought it was a mistake, so if for You it is better, for me it is also ok.

**Comment by cajun-rat** [comment]
Cool. Thank you!

**Pull request merged**